src/lib/multiaddr.ts

```diff
@@ -77,3 +81,34 @@ export const getRandomRelayId = (relays: Relays): string => {
 
   return randomRelayId;
 };
+
+export const getEvenlyDistributedIds = (
+  relays: Relays,
+  count = 1
+): Array<string> => {
+  const addrs = resolveAddrs(relays);
+  const ids = getIds(addrs);
+  return getEvenlyDistributedIdsFromTheList(ids, count);
+};
+
+const offsets = new Map<string, number>();
+
+export const getEvenlyDistributedIdsFromTheList = (
```

Comment:
I think it could be written simpler if you make a function which generates a list of randomly distributed indices. Something like:

```
const evenlySpreadIndices = (count, total) => 
[...Array(count).keys()]
   .map(x => x + 1)
   .map(x => total / x))
   .map(x => x - 1)
   .map(Math.round);
```

And then use these indices to pick corresponding items from the collection

Reply:
Not completely sure what you are suggesting, exactly.
In my implementation, I also have this offset. This way things can be distributed real evenly across deploys,
e.g. without that, each deploy will have the same peer ID as the first peer ID

Reply:
Could you please add more comments about the algorithm, namely

1. Describe what is the algorithm and why do we need it
2. Add a note that the function is not pure, and what side effects are

Reply:
Done